_api_app/app/Sites/Sections/Entries/SectionEntriesDataService.php

```diff
@@ -509,6 +511,164 @@ public function delete()
+            'mediaCacheData' => [
+                '@attributes'=> [
+                    'type' => $defaultGalleryType,
+                    'fullscreen' => $galleryFullScreen
+                ],
+                'file' => []
+            ]
+        ];
+
+        if ($tag) {
+            $sectionTagsDataService = new SectionTagsDataService($this->SITE, $this->SECTION_NAME);
+            $tags = $sectionTagsDataService->getSectionTagsState();
+
+            // Find tag and set tag title for entry
+            if ($tags['tag']) {
+                $tag_key = array_search(
+                    $tag,
+                    array_column(
+                        array_column(
+                            $tags['tag'],
+                            '@attributes'
+                        ),
+                        'name'
+                    )
+                );
+
+                if ($tag_key !== false) {
+                    $tag_title = $tags['tag'][$tag_key]['@value'];
+
+                    $new_entry['tags'] = [
+                        'tag' => $tag_title
+                    ];
+                }
+            }
+        }
+
+        // Insert entry in correct position
+        $entry_order = count($entries['entry']);
+        if ($before_entry) {
+            $order = array_search($before_entry, array_column($entries['entry'], 'id'));
+
+            if ($order !== false) {
+                $entry_order = $order;
+                array_splice($entries['entry'], $order, 0, [$new_entry]);
+            } else {
+                $entries['entry'][] = $new_entry;
+            }
+
+        } else {
+            $entries['entry'][] = $new_entry;
+        }
+
+        // Save sorted entries
+        $this->array2xmlFile($entries, $this->XML_FILE, $this->ROOT_ELEMENT);
+
+        // Add params for frontend state
+        $new_entry['sectionName'] = $this->SECTION_NAME;
+        $new_entry['order'] = $entry_order;
+
+        // Update section entry count
+        $siteSectionsDataService = new SiteSectionsDataService($this->SITE);
+        $sections = $siteSectionsDataService->get();
+        $section_order = array_search($this->SECTION_NAME, array_column($sections, 'name'));
+        $section_entry_count = count($entries['entry']);
+        $siteSectionsDataService->saveValueByPath(
+            implode('/', [
+                $this->SITE,
+                'section',
+                $section_order,
+                '@attributes',
+                'entry_count'
+            ]),
+            $section_entry_count
+        );
+
+        // update direct content property
+        $siteSectionsDataService = new SiteSectionsDataService($this->SITE);
+        $sectionTagsDataService = new SectionTagsDataService($this->SITE, $this->SECTION_NAME);
+        $section_tags = $sectionTagsDataService->populateTags();
+        $has_direct_content = !$section_tags['allHaveTags'] ? 1 : 0;
+        $siteSectionsDataService->saveValueByPath(
```

Comment:
Why not use the same `SiteSectionsDataService`?

Reply:
Added a comment why: https://github.com/berta-cms/berta/pull/225/files#diff-94bb9a8e67f5f1a90c376f66f4ea67baR634

The same in `deleteEntry` method.